src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -669,6 +670,8 @@ public function load(array $configs, ContainerBuilder $container)
             ->addTag('serializer.normalizer');
         $container->registerForAutoconfiguration(ConstraintValidatorInterface::class)
             ->addTag('validator.constraint_validator');
+        $container->registerForAutoconfiguration(GroupSequenceProviderInterface::class)
```

Comment:
Using the same interface for external provider and self-provided implementation has a big drawback: all your DTOs implementing the interface for self usage will be collected with this tag and kept in the container, forcing you to manually exclude them if you don't want to face issues where the method would be called on a dummy instance instead of the DTO itself.

Reply:
Let me know if I'm missing something, but without specifying the new `class` option, it's not possible to call an external provider or any other than the one defined in the `class` option. The validator should always call the underlying DTO method in that cases.

In the other hand, there shouldn't be too many cases to be concerned about regarding the number of useless providers in this container, so I would keep them in favor of better DX.

Reply:
The issue with useless DTOs in the container is that any non-autowirable parameter in the constructor of those DTOs become a compilation error for the container (a removed service does not trigger an autowiring error, but preserved services must be valid).

Reply:
Anyway, if you want to pass the object to the method as discussed in https://github.com/symfony/symfony/pull/51348#discussion_r1297090934, you will need a different interface.

Reply:
Agreed, let's add a new interface then. Thanks for the feedback!